Docs/introducing-blazor-forms.md

```diff
@@ -0,0 +1,215 @@
+| [`InputNumber<TValue>`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.forms.inputnumber-1?view=aspnetcore-5.0)     | `<input type="number">`   |
+| [`InputRadio`](https://docs.microsoft.com/en-us/aspnet/core/blazor/forms-validation?view=aspnetcore-5.0#radio-buttons)                           | `<input type="radio">`    |
+| [`InputRadioGroup`](https://docs.microsoft.com/en-us/aspnet/core/blazor/forms-validation?view=aspnetcore-5.0#radio-buttons)                      | `<input type="radio">`    |
+| [`InputSelect<TValue>`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.forms.inputselect-1?view=aspnetcore-5.0)     | `<select>`                |
+| [`InputText`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.forms.inputtext?view=aspnetcore-5.0)                   | `<input>`                 |
+| [`InputTextArea`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.forms.inputtextarea?view=aspnetcore-5.0)           | `<textarea>`              |
+
+
+All of these input components provide two-way data binding which makes it easier to keep our UI components and the data objects in sync especially with form validations. Moreover, we can also specify any additional attributes that are not provided by Blazor, which will be rendered to the to the HTML page.
+
+We also have the opportunity to use any of the third-party specialized controls that are available via NuGet to extend the functionality of our forms.  These include specialized components such as:
+
+* [Blazored](https://github.com/Blazored) component library which includes controls that allow users to enter and view Markdown in forms
+* [Telerik](https://www.telerik.com/blazor-ui)'s Blazor component is a comprehensive library of components that can be used to create very customized solutions
+* [MudBlazor](https://mudblazor.com/) component library inspired by Google's Material design language and written entirely in C#
+
+## Model Binding
+
+When creating a new EditForm in Blazor, we must pass either a [Model](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.forms.editform.model?view=aspnetcore-5.0#Microsoft_AspNetCore_Components_Forms_EditForm_Model) or an [EditContext](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.forms.editcontext?view=aspnetcore-5.0) parameter into it. The descriptions of the two are as follows:
+
+It is worth noting that you cannot pass both at the same time as it will throw an error if you do so.
+
+Let's see how the EditForm gets created under the covers. The simplified version of this process is shown in the diagram below.
+
+![](./images/creation-of-editform.png)
+
+Now that we have an idea of how the EditForm is created, let's see how Blazor handles when you type in something in an input text component.
```

Comment:
Somewhere in here, I suggest that we explain to the reader why we are taking them through all of this.  Why are we explaining the component lifecycle to them?  How can it benefit them?